prqlc/bindings/prqlc-python/src/lib.rs

```diff
@@ -1,44 +1,43 @@
 #![cfg(not(target_family = "wasm"))]
 use std::str::FromStr;
 
+use prqlc_lib::ErrorMessages;
 use pyo3::{exceptions, prelude::*};
 
 #[pyfunction]
 pub fn compile(prql_query: &str, options: Option<CompileOptions>) -> PyResult<String> {
-    let options = options.map(convert_options).transpose();
-
-    options
-        .and_then(|opts| {
-            Ok(prql_query)
-                .and_then(prqlc_lib::prql_to_pl)
-                .and_then(prqlc_lib::pl_to_rq)
-                .and_then(|rq| prqlc_lib::rq_to_sql(rq, &opts.unwrap_or_default()))
-        })
+    let Ok(options) = options.map(convert_options).transpose() else {
+        return Err(PyErr::new::<exceptions::PyValueError, _>(
+            "Invalid options".to_string(),
+        ));
+    };
+
+    Ok(prql_query)
+        .and_then(prqlc_lib::prql_to_pl)
+        .and_then(prqlc_lib::pl_to_rq)
+        .and_then(|rq| prqlc_lib::rq_to_sql(rq, &options.unwrap_or_default()))
         .map_err(|e| e.composed(&prql_query.into()))
-        .map_err(|e| (PyErr::new::<exceptions::PySyntaxError, _>(e.to_string())))
+        .map_err(|e| (PyErr::new::<exceptions::PyValueError, _>(e.to_string())))
```

Comment:
Sorry to ask this now, but is the `display` argument really effective here?

I had trouble getting this option to work when I ported it to prqlc-r (PRQL/prqlc-r#317), but after looking at the source code I realized that this only works within the `prql::compile` function.
(Sorry don't have time to try today, but if no one sees this I guess I can take a look in a few days)

Reply:
Yes, I'm not sure why I did it that way, I guess I was adjusting from the previous version. Please feel free to simplify!

Reply:
#4919